src/elements/earn/portfolio/v3/holdings/V3HoldingPage.tsx

```diff
@@ -0,0 +1,307 @@
+import { useNavigation } from 'hooks/useNavigation';
+import { useParams } from 'react-router-dom';
+import { ReactComponent as IconChevronRight } from 'assets/icons/chevronRight.svg';
+import { ReactComponent as IconWarning } from 'assets/icons/warning.svg';
+import { Image } from 'components/image/Image';
+import { useAppSelector } from 'store';
+import {
+  getPortfolioHoldings,
+  getIsLoadingHoldings,
+} from 'store/portfolio/v3Portfolio';
+import { PopoverV3 } from 'components/popover/PopoverV3';
+import { prettifyNumber, toBigNumber } from 'utils/helperFunctions';
+import { Button, ButtonSize, ButtonVariant } from 'components/button/Button';
+import { useState } from 'react';
+import { bntToken } from 'services/web3/config';
+import { ContractsApi } from 'services/web3/v3/contractsApi';
+import { fetchWithdrawalRequestOutputBreakdown } from 'services/web3/v3/portfolio/withdraw';
+import useAsyncEffect from 'use-async-effect';
+import { expandToken, shrinkToken } from 'utils/formulas';
+import { DepositV3Modal } from 'elements/earn/pools/poolsTable/v3/DepositV3Modal';
+import V3WithdrawModal from '../initWithdraw/V3WithdrawModal';
+import BigNumber from 'bignumber.js';
+import { Token } from 'services/observables/tokens';
+import { getTokenById } from 'store/bancor/bancor';
+import { getV3byID } from 'store/bancor/pool';
+import { PoolV3 } from 'services/observables/pools';
+import { WalletConnectRequest } from 'elements/walletConnect/WalletConnectRequest';
+import { V3ManageProgramsModal } from './V3ManageProgramsModal';
+
+export const V3HoldingPage = () => {
+  const { id } = useParams();
+  const { goToPage } = useNavigation();
+  const [isOpen, setIsOpen] = useState(false);
+
+  const account = useAppSelector((state) => state.user.account);
+  const holdings = useAppSelector(getPortfolioHoldings);
+  const isLoadingHoldings = useAppSelector(getIsLoadingHoldings);
+  const holding = holdings.find((x) => x.pool.poolDltId === id);
+  const token = useAppSelector<Token | undefined>((state: any) =>
```

Comment:
I fixed the type inference of useAppSelector a while ago, you dont need to type cast the return value anymore (<Token | undefined>)

and state: any needs to be removed for the inferred types to work.

Reply:
Awesome!